spacy_methods.py

```diff
@@ -0,0 +1,213 @@
+                    sentence_end = (
+                        ent.sent.end - 1
+                    )  # Index of the last word in the sentence
+                    sentence_index = ent.start - sentence_start
+                    word_index = word_count + start_index
+
+                    specific_entities.append(
+                        [
+                            ent.text,
+                            ent.label_,
+                            start_index,
+                            end_index,
+                            sentence,
+                            sentence_count,
+                            sentence_start,
+                            sentence_end,
+                            word_index,
+                        ]
+                    )
+
+            # Adds the sentence_end indexing to the total word count
+        word_count += len(doc)
+    return specific_entities
+
+
+entities = get_specific_entities(sentences)
+
+# raw_entity_list = list(entities)
+
+# Counts the number of distinct times an entity appears in the article.
+def entity_counter(lst):
+
+    raw_entity_list = list(entities)
+    count_dict = {}
+
+    # Index the list elements to a dictionary and increment their counts
+    for i, item in enumerate(lst):
+
+        entity_sentence_check = "{}".format(item[0])
+        if entity_sentence_check in count_dict:
+            count_dict[entity_sentence_check] += 1
+        else:
+            count_dict[entity_sentence_check] = 1
+    return count_dict
+
+
+duplicate_items = entity_counter(entities)
+
+# Algorithm to assign new key values to 1, increment if already existing
+def append_to_array(key, value, dict_to_check):
+    if key in dict_to_check:
+        dict_to_check[key].append(value)
+    else:
+        dict_to_check[key] = [value]
+
+
+raw_entity_list = list(entities)
+
+
+def entity_indexer(lst):
+    ent_index_dict = {}
+
+    for i, entity_item in enumerate(lst):
+        entity_sentence_check = "{}".format(entity_item[0])
+
+        # entity_item[0] - Entity's name as a string
+        # entity_item[5] - the sentence number index
+
+        append_to_array(entity_item[0], entity_item[5], ent_index_dict)
+    return ent_index_dict
+
+
+ent_sentence_index = entity_indexer(raw_entity_list)
+
+
+def verb_matcher(txt):
+    # Verb Finder with Matcher
+    verb_matcher = Matcher(nlp.vocab)
+    verb_pattern = [{"POS": "VERB", "OP": "+"}]
+    verb_matcher.add("VERBS", [verb_pattern])
+    matches = verb_matcher(txt)
+
+    verb_information = []
+
+    for match in matches:
+
+        word_index = match[1]
+        original_verb = doc[match[1]]
+        lemmatized_verb = doc[match[1]].lemma_
+        verb_information.append([word_index, original_verb, lemmatized_verb])
+
+    return verb_information
+
+
+verbs = verb_matcher(doc)
+# print(verbs)
+
+# List of Verbs
+verbs = verb_matcher(doc)
+
+# List of Article's Sentences
+
+sentences = sentence_generator(doc)
```

Comment:
Same comment about moving it under `if __name__ == "__main__":`

Reply:
Moved. Resolved.